src/main/java/org/symphonyoss/symphony/messageml/MessageMLParser.java

```diff
@@ -435,6 +439,18 @@ public Element createElement(org.w3c.dom.Element element, Element parent) throws
     }
   }
 
+  /**
+   * Loads the values of the "id" attribute of elements being parsed and verifies if these values are unique.
+   *
+   * @param id
+   * @throws InvalidInputException
+   */
+  public void loadElementId(String id) throws InvalidInputException {
+    if (!elementIds.add(id)) {
+      throw new InvalidInputException("Elements must have unique ids.");
```

Comment:
I would mention the id that is duplicated. This error message goes back to the user calling.